dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/step/validation/CclicenseValidator.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.submit.step.validation;
+
+import static org.dspace.app.rest.repository.WorkspaceItemRestRepository.OPERATION_PATH_SECTIONS;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+
+import org.dspace.app.rest.model.ErrorRest;
+import org.dspace.app.rest.submit.SubmissionService;
+import org.dspace.app.util.DCInputsReaderException;
+import org.dspace.app.util.SubmissionStepConfig;
+import org.dspace.content.InProgressSubmission;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.license.CreativeCommonsServiceImpl;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This class validates that the Creative Commons License has been granted for the
+ * in-progress submission.
+ *
+ * @author Mattia Vianelli (Mattia.Vianelli@4science.com)
+ */
+public class CclicenseValidator extends AbstractValidation {
+
+    /**
+     * Construct a Creative Commons License configuration.
+     * @param configurationService DSpace configuration provided by the DI container.
+     */
+    @Inject
+    public CclicenseValidator(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    private final ConfigurationService configurationService;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private CreativeCommonsServiceImpl creativeCommonsService;
+
+    public static final String ERROR_VALIDATION_CCLICENSEREQUIRED = "error.validation.cclicense.required";
+
+    private String name;
+
+    private Boolean required;
```

Comment:
Please remove this property since we are relying directly on the `configurationService` to retrieve values.